Content.Client/HealthAnalyzer/UI/HealthAnalyzerWindow.xaml.cs

```diff
@@ -1,76 +1,203 @@
+                var groupContainer = new BoxContainer
+                {
+                    Margin = new Thickness(0, 0, 30, 0),
+                    Align = BoxContainer.AlignMode.Begin,
+                    Orientation = BoxContainer.LayoutOrientation.Vertical
+                };
 
-                HashSet<string> shownTypes = new();
+                groupContainer.AddChild(CreateDiagnosticGroupTitle(groupTitleText, damageGroupId, (int) damageAmount));
+                groupContainer.AddChild(new PanelContainer { StyleClasses = { "LowDivider" } });
 
-                var protos = IoCManager.Resolve<IPrototypeManager>();
+                // Show the damage for each type in that group.
+                var group = _prototypes.Index<DamageGroupPrototype>(damageGroupId);
 
-                // Show the total damage and type breakdown for each damage group.
-                foreach (var (damageGroupId, damageAmount) in damagePerGroup)
+                foreach (var type in group.DamageTypes)
                 {
-                    text.Append($"\n{Loc.GetString("health-analyzer-window-damage-group-text", ("damageGroup", Loc.GetString("health-analyzer-window-damage-group-" + damageGroupId)), ("amount", damageAmount))}");
-                    // Show the damage for each type in that group.
-                    var group = protos.Index<DamageGroupPrototype>(damageGroupId);
-                    foreach (var type in group.DamageTypes)
+                    if (damageDict.TryGetValue(type, out var typeAmount))
                     {
-                        if (damagePerType.TryGetValue(type, out var typeAmount))
+                        // If damage types are allowed to belong to more than one damage group,
+                        // they may appear twice here. Mark them as duplicate.
+                        if (!shownTypes.Contains(type))
                         {
-                            // If damage types are allowed to belong to more than one damage group, they may appear twice here. Mark them as duplicate.
-                            if (!shownTypes.Contains(type))
-                            {
-                                shownTypes.Add(type);
-                                text.Append($"\n- {Loc.GetString("health-analyzer-window-damage-type-text", ("damageType", Loc.GetString("health-analyzer-window-damage-type-" + type)), ("amount", typeAmount))}");
-                            }
+                            shownTypes.Add(type);
+
+                            var damageString = Loc.GetString(
+                                "health-analyzer-window-damage-type-text",
+                                ("damageType", Loc.GetString("health-analyzer-window-damage-type-" + type)),
+                                ("amount", typeAmount)
+                            );
+
+                            groupContainer.AddChild(CreateDiagnosticItemLabel(damageString, (int) typeAmount == 0));
```

Comment:
This cast is unnecessary as it's already fixedpoint2 and fixedpoint2.zero exists.